refined storage with InvMove will crash at launch

### Describe the bug

installed refined storage 2 with InvMove, crashed at launch with error code -1

### How can we reproduce this bug or crash?

install refined storage 2 with InvMove with fabric, and launch

### What Minecraft version is this happening on?

Minecraft 1.20.1

### What Forge or Fabric version is this happening on?

fabric 0.15.3

### What Refined Storage version is this happening on?

version 2.0 milestone 3.2

### Relevant log output

```shell
---- Minecraft Crash Report ----
// I bet Cylons wouldn't have this problem.

Time: 2024-01-01 13:11:59
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'invmove'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at me.pieking1215.invmove.InvMoveConfig.readUnrecognizedConfig(InvMoveConfig.java:455)
	at me.pieking1215.invmove.InvMoveConfig.load(InvMoveConfig.java:332)
	at me.pieking1215.invmove.InvMove.finishInit(InvMove.java:111)
	at me.pieking1215.invmove.fabric.InvMoveFabric.onInitializeClient(InvMoveFabric.java:57)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 9 more
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.class_315.method_42437()" because "this.field_1690" is null
	at net.minecraft.class_310.method_1573(class_310.java:924)
	at com.refinedmods.refinedstorage2.platform.common.support.tooltip.SmallText.getScale(SmallText.java:14)
	at com.refinedmods.refinedstorage2.platform.common.support.tooltip.SmallTextClientTooltipComponent.<init>(SmallTextClientTooltipComponent.java:15)
	at com.refinedmods.refinedstorage2.platform.common.support.AbstractBaseScreen.<clinit>(AbstractBaseScreen.java:43)
	... 16 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:458)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.23110.3 (9ebb3643d539a6fc4659898b1df3124d5da4c0a9):Microsoft Corporation
		NLAapi.dll:Network Location Awareness 2:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		OleAut32.dll:OLEAUT32.DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Wldp.dll:Windows Lockdown Policy:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		apphelp.dll:Application Compatibility Client Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		awt.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:DHCP Client Service:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:DHCPv6 Client:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.19041.3758 (WinBuild.160101.0800):Microsoft Corporation
		glfw.dll:GLFW 3.4.0 DLL:3.4.0:GLFW
		java.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		javaw.exe:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jemalloc.dll
		jimage.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jli.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jna6245479072926981961.dll:JNA native library:6.1.4:Java(TM) Native Access (JNA)
		jsvml.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		jvm.dll:Java HotSpot(TM) 64-Bit server VM:17.0.9.0:Oracle Corporation
		kernel.appcore.dll:AppModel API Host:10.0.19041.3758 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		management_ext.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		msvcp140.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		nio.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		ntdll.dll:NT Layer DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP Name Space Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.31.31103.0:Microsoft Corporation
		verify.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
		win32u.dll:Win32u:10.0.19041.3803 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.19041.3636 (WinBuild.160101.0800):Microsoft Corporation
		wshunix.dll:AF_UNIX Winsock2 Helper DLL:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:Java(TM) Platform SE binary:17.0.9.0:Oracle Corporation
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:211)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)

-- System Details --
Details:
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.9, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 621456048 bytes (592 MiB) / 1398800384 bytes (1334 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 16
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 7 5800X 8-Core Processor             
	Identifier: AuthenticAMD Family 25 Model 33 Stepping 2
	Microarchitecture: Zen 3
	Frequency (GHz): 3.79
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 16
	Graphics card #0 name: NVIDIA GeForce RTX 3070 Ti
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x2482
	Graphics card #0 versionInfo: DriverVersion=31.0.15.4633
	Memory slot #0 capacity (MB): 16384.00
	Memory slot #0 clockSpeed (GHz): 2.67
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 16384.00
	Memory slot #1 clockSpeed (GHz): 2.67
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 37515.39
	Virtual memory used (MB): 14529.88
	Swap memory total (MB): 4864.00
	Swap memory used (MB): 88.87
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx4096m
	Fabric Mods: 
		advancementinfo: AdvancementInfo 1.20-fabric0.83.0-1.4
		appleskin: AppleSkin 2.5.1+mc1.20
		architectury: Architectury 9.1.12
		authme: Auth Me 7.0.2+1.20
		auudio: Auudio 1.0.3
		balm-fabric: Balm 7.2.1
		bankstorage: Bank Storage 0.3.8
		bedrockify: BedrockIfy 1.9.1+mc1.20
			panorama_screens: Panorama Screens 1.0+fabric+mc1.20
		betterpingdisplay: Better Ping Display 1.1.1
		betterthirdperson: Better Third Person 1.9.0
		bookshelf: Bookshelf 20.1.8
		borderlessmining: Borderless Mining 1.1.8+1.20.1
		bwncr: Bad Wither No Cookie Reloaded 3.17.0
		cavedust: Cave Dust 1.4.1
			kirin: Kirin UI 1.15.0
		cherishedworlds: Cherished Worlds 6.1.5+1.20.1
		cleancut: CleanCut 6.0+mc1.19.1-mc1.20.x
		cleardespawn: Clear Despawn 1.1.15
		cloth-config: Cloth Config v11 11.1.118
			cloth-basic-math: cloth-basic-math 0.6.1
		clumps: Clumps 12.0.0.3
		cobblemon: Cobblemon 1.4.1+1.20.1
		collective: Collective 7.26
		continuity: Continuity 3.0.0-beta.4+1.20.1
		craftingtweaks: Crafting Tweaks 18.2.2
		craftpresence: CraftPresence 2.2.6
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.9+132c48c177
		ctrl-q: Ctrl Q 1.9
		dismountentity: Dismount Entity 3.2
		dynamic_fps: Dynamic FPS 3.3.1
		enchdesc: EnchantmentDescriptions 17.0.11
		entityculling: EntityCulling-Fabric 1.6.2-mc1.20.1
		fabric-api: Fabric API 0.91.0+1.20.1
			fabric-api-base: Fabric API Base 0.4.30+7abfd51577
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.35+4d8536c977
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.12+215bbe9677
			fabric-block-api-v1: Fabric Block API (v1) 1.0.10+92a0d36777
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.0+92a0d36777
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.40+b3afc78b77
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.1+97bb207577
			fabric-command-api-v1: Fabric Command API (v1) 1.2.33+f71b366f77
			fabric-command-api-v2: Fabric Command API (v2) 2.2.12+b3afc78b77
			fabric-commands-v0: Fabric Commands (v0) 0.2.50+df3654b377
			fabric-containers-v0: Fabric Containers (v0) 0.1.63+df3654b377
			fabric-content-registries-v0: Fabric Content Registries (v0) 4.0.10+57aed33f77
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.4+a1a980da77
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.18+aeb40ebe77
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 12.3.3+d7b148e077
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.53+8536527b77
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.22+b3afc78b77
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.6.1+e91849a877
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.62+df3654b377
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.39+ae9f657a77
			fabric-item-api-v1: Fabric Item API (v1) 2.1.27+b3afc78b77
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.11+d7b148e077
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.36+fb8d95da77
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.34+df3654b377
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.21+b3afc78b77
			fabric-loot-api-v2: Fabric Loot API (v2) 1.2.0+96dfa95977
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.44+9e7660c677
			fabric-message-api-v1: Fabric Message API (v1) 5.1.8+d7b148e077
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.49+b3afc78b77
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 1.0.2+709a987177
			fabric-models-v0: Fabric Models (v0) 0.4.1+9386d8a777
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.10+eeb8eb3677
			fabric-networking-v0: Fabric Networking (v0) 0.3.50+df3654b377
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 11.1.2+4ee0bc6077
			fabric-particles-v1: Fabric Particles (v1) 1.1.1+201a23a077
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.20+b3afc78b77
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.3.2+4df89eb277
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.2.0+39a511ba77
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.5.0+39a511ba77
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.45+df3654b377
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.36+92a0d36777
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.27+b3afc78b77
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.48+df3654b377
			fabric-rendering-v1: Fabric Rendering (v1) 3.0.7+b3afc78b77
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.7+29de845d77
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.7+b3afc78b77
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.29+b3afc78b77
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.12+b3afc78b77
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.3.3+c81d263177
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 4.3.0+6c31357e77
		fabric-language-kotlin: Fabric Language Kotlin 1.10.17+kotlin.1.9.22
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.9.22
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.9.22
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.23.1
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.7.3
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.7.3
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.5.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.6.2
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.6.2
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.6.2
		fabricloader: Fabric Loader 0.15.3
			mixinextras: MixinExtras 0.3.2
		fallingleaves: Falling Leaves 1.15.4
		fancymenu: FancyMenu 2.14.10
		faster_entity_animations: FastAnim 1.6
		ferritecore: FerriteCore 6.0.1
		feytweaks: FeyTweaks 1.20-1.2.7
			midnightlib: MidnightLib 1.4.1
		fmextension_audio: FancyMenu Audio Extension 1.1.2
		forgeconfigapiport: Forge Config API Port 8.0.0
		highlight: Highlight 2.0.1
			resourcefullib: Resourceful Lib 2.0.7
				com_teamresourceful_yabn: yabn 1.0.3
		highlighter: Highlighter 1.1.6
		iceberg: Iceberg 1.1.18
		immediatelyfast: ImmediatelyFast 1.2.8+1.20.4
			net_lenni0451_reflect: Reflect 1.3.1
		indium: Indium 1.0.27+mc1.20.1
		invmove: InvMove 0.8.4
		iris: Iris 1.6.11
			io_github_douira_glsl-transformer: glsl-transformer 2.0.0-pre13
			org_anarres_jcpp: jcpp 1.4.14
			org_antlr_antlr4-runtime: antlr4-runtime 4.11.1
		java: Java HotSpot(TM) 64-Bit Server VM 17
		konkrete: Konkrete 1.7.0
		krypton: Krypton 0.2.3
			com_velocitypowered_velocity-native: velocity-native 3.2.0-SNAPSHOT
		languagereload: Language Reload 1.5.10+1.20.1
		lazydfu: LazyDFU 0.1.3
		letmedespawn: Let Me Despawn 1.2.0
		lithium: Lithium 0.11.2
		loadmyresources: Load My Resources 1.0.4
		longerchathistory: Longer Chat History 1.3
		macos_input_fixes: MacOS Input Fixes 1.6
		make_bubbles_pop: Make Bubbles Pop Mod 0.2.0-fabric
		memoryleakfix: Memory Leak Fix 1.1.2
		minecraft: Minecraft 1.20.1
		modmenu: Mod Menu 7.2.2
		monsters_in_the_closet: Monsters in the Closet 1.0.3+1.20
		moreculling: More Culling 1.20.1-0.19.0
			conditional-mixin: conditional mixin 0.3.2
		netherportalfix: NetherPortalFix 13.0.1
		notenoughanimations: NotEnoughAnimations 1.6.4
		presencefootsteps: Presence Footsteps 1.9.0
		reeses-sodium-options: Reese's Sodium Options 1.6.5+mc1.20.1-build.95
		refinedstorage2: Refined Storage 2 2.0.0-milestone.3.2
			com_refinedmods_refinedstorage2_refinedstorage2-query-parser_: refinedstorage2-query-parser 2.0.0-milestone.3.2
			noindium: No Indium? 1.1.0+1.20
			refinedstorage2-core-api: Refined Storage 2 Core API 2.0.0-milestone.3.2
			refinedstorage2-grid-api: Refined Storage 2 Grid API 2.0.0-milestone.3.2
			refinedstorage2-network: Refined Storage 2 Network 2.0.0-milestone.3.2
			refinedstorage2-network-api: Refined Storage 2 Network API 2.0.0-milestone.3.2
			refinedstorage2-resource-api: Refined Storage 2 Resource API 2.0.0-milestone.3.2
			refinedstorage2-storage-api: Refined Storage 2 Storage API 2.0.0-milestone.3.2
			team_reborn_energy: Energy 3.0.0
		roughlyenoughitems: Roughly Enough Items 12.0.684
			error_notifier: Error Notifier 1.0.9
		roughlyenoughprofessions: Roughly Enough Professions 2.0.2
		roughlysearchable: Roughly Searchable 2.6.0+1.20.1
		showmeyourskin: Show Me Your Skin! 1.6.11+1.20
			cardinal-components-base: Cardinal Components API (base) 5.2.1
			cardinal-components-entity: Cardinal Components API (entities) 5.2.1
			cicada: CICADA 0.4.0
		shulkerboxtooltip: Shulker Box Tooltip 4.0.4+1.20.1
		slimyfloor: Slimy Floor 1.20-1.2.8
		sodium: Sodium 0.5.3
		sodium-extra: Sodium Extra 0.5.1+mc1.20.1-build.112
			caffeineconfig: CaffeineConfig 1.3.0+1.17
			crowdin-translate: CrowdinTranslate 1.4+1.19.3
		sspb: Sodium Shadowy Path Blocks 3.2.1
		starlight: Starlight 1.1.2+fabric.dbc156f
		stendhal: Stendhal 1.4.1-1.20
			libgui: LibGui 8.0.1+1.20
				jankson: Jankson 5.0.1+j1.2.2
					blue_endless_jankson: jankson 1.2.2
				libninepatch: LibNinePatch 1.2.0
		superflatworldnoslimes: Superflat World No Slimes 3.2
		threadtweak: ThreadTweak 1.20.2-0.1.1
		titlefixer: Title Fixer 1.0+1.20.1
		toolstats: ToolStats 16.0.7
		tooltipfix: ToolTip Fix 1.1.1-1.20
		visuality: Visuality 0.7.1+1.20
		waterdripsound: Drip Sounds 1.19-0.3.2
		waveycapes: WaveyCapes 1.3.2
		xaerominimap: Xaero's Minimap 23.9.3
		xaeroworldmap: Xaero's World Map 1.37.2
		yet_another_config_lib_v3: YetAnotherConfigLib 3.2.1+1.20
			com_twelvemonkeys_common_common-image: common-image 3.10.0-SNAPSHOT
			com_twelvemonkeys_common_common-io: common-io 3.10.0-SNAPSHOT
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0-SNAPSHOT
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0-SNAPSHOT
			org_quiltmc_parsers_gson: gson 0.2.1
			org_quiltmc_parsers_json: json 0.2.1
		yosbr: YOSBR 0.1.2
		zoomify: Zoomify 2.11.2
			com_akuleshov7_ktoml-core-jvm: ktoml-core-jvm 0.4.1
			dev_isxander_settxi_settxi-core: settxi-core 2.10.6
			dev_isxander_settxi_settxi-kotlinx-serialization: settxi-kotlinx-serialization 2.10.6
	Launched Version: 1.20.1
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Added
 
-- Ported to Minecraft 1.20.4.
+-   Ported to Minecraft 1.20.4.
 -   Custom disk models. Fluid disks now have a different model.
 -   Portable Grid
 
@@ -21,6 +21,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 ### Fixed
 
 -   Fixed bug where Grid contents weren't synced properly when a network merge occurs.
+-   Fixed incompatibility crash with InvMove on Fabric.
 
 ## [2.0.0-milestone.3.2] - 2023-11-03
 

--- a/refinedstorage2-network/src/main/java/com/refinedmods/refinedstorage2/api/network/impl/energy/AbstractProxyEnergyStorage.java
+++ b/refinedstorage2-network/src/main/java/com/refinedmods/refinedstorage2/api/network/impl/energy/AbstractProxyEnergyStorage.java
@@ -6,7 +6,7 @@ import com.refinedmods.refinedstorage2.api.network.energy.EnergyStorage;
 public abstract class AbstractProxyEnergyStorage implements EnergyStorage {
     private final EnergyStorage energyStorage;
 
-    public AbstractProxyEnergyStorage(final EnergyStorage energyStorage) {
+    protected AbstractProxyEnergyStorage(final EnergyStorage energyStorage) {
         this.energyStorage = energyStorage;
     }
 

--- a/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/support/tooltip/SmallTextClientTooltipComponent.java
+++ b/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/support/tooltip/SmallTextClientTooltipComponent.java
@@ -8,11 +8,9 @@ import org.joml.Matrix4f;
 
 public class SmallTextClientTooltipComponent implements ClientTooltipComponent {
     private final Component text;
-    private final float scale;
 
     public SmallTextClientTooltipComponent(final Component text) {
         this.text = text;
-        this.scale = SmallText.getScale();
     }
 
     @Override
@@ -21,6 +19,7 @@ public class SmallTextClientTooltipComponent implements ClientTooltipComponent {
                            final int y,
                            final Matrix4f pose,
                            final MultiBufferSource.BufferSource buffer) {
+        final float scale = SmallText.getScale();
         SmallText.render(font, text.getVisualOrderText(), x, y, scale, pose, buffer);
     }
 
@@ -31,6 +30,7 @@ public class SmallTextClientTooltipComponent implements ClientTooltipComponent {
 
     @Override
     public int getWidth(final Font font) {
+        final float scale = SmallText.getScale();
         return (int) (font.width(text) * scale);
     }
 }

--- a/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/upgrade/UpgradeWithEnchantedBookRecipe.java
+++ b/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/upgrade/UpgradeWithEnchantedBookRecipe.java
@@ -4,21 +4,16 @@ import com.refinedmods.refinedstorage2.platform.common.content.Items;
 
 import java.util.Objects;
 import java.util.Optional;
-import javax.annotation.Nullable;
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
-import net.minecraft.core.Holder;
 import net.minecraft.core.NonNullList;
-import net.minecraft.core.registries.Registries;
+import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.ListTag;
-import net.minecraft.resources.RegistryFixedCodec;
-import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.inventory.CraftingContainer;
 import net.minecraft.world.item.EnchantedBookItem;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.crafting.CraftingBookCategory;
 import net.minecraft.world.item.crafting.Ingredient;
@@ -30,29 +25,36 @@ import net.minecraft.world.item.enchantment.EnchantmentInstance;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
 
+import static java.util.Objects.requireNonNull;
+
 public class UpgradeWithEnchantedBookRecipe extends ShapedRecipe {
     public static final Codec<UpgradeWithEnchantedBookRecipe> CODEC = RecordCodecBuilder.create(
         instance -> instance.group(
-            RegistryFixedCodec.create(Registries.ENCHANTMENT).fieldOf("enchantment")
-                .forGetter(UpgradeWithEnchantedBookRecipe::getEnchantment),
+            Codec.STRING.fieldOf("enchantment")
+                .xmap(ResourceLocation::new, ResourceLocation::toString)
+                .forGetter(UpgradeWithEnchantedBookRecipe::getEnchantmentId),
             Codec.INT.fieldOf("level").orElse(1)
                 .forGetter(UpgradeWithEnchantedBookRecipe::getEnchantmentLevel),
-            RegistryFixedCodec.create(Registries.ITEM).fieldOf("result")
-                .forGetter(UpgradeWithEnchantedBookRecipe::getResultItem)
+            Codec.STRING.fieldOf("result")
+                .xmap(ResourceLocation::new, ResourceLocation::toString)
+                .forGetter(UpgradeWithEnchantedBookRecipe::getResultItemId)
         ).apply(instance, UpgradeWithEnchantedBookRecipe::new)
     );
 
-    private final Holder<Enchantment> enchantment;
+    private final ResourceLocation enchantmentId;
     private final int level;
-    private final Holder<Item> resultItem;
+    private final ResourceLocation resultItemId;
 
-    UpgradeWithEnchantedBookRecipe(final Holder<Enchantment> enchantment,
+    UpgradeWithEnchantedBookRecipe(final ResourceLocation enchantmentId,
                                    final int level,
-                                   final Holder<Item> resultItem) {
+                                   final ResourceLocation resultItemId) {
         super("", CraftingBookCategory.MISC, new ShapedRecipePattern(3, 3, NonNullList.of(
             Ingredient.EMPTY,
             Ingredient.of(new ItemStack(Items.INSTANCE.getQuartzEnrichedIron())),
-            Ingredient.of(EnchantedBookItem.createForEnchantment(new EnchantmentInstance(enchantment.value(), level))),
+            Ingredient.of(EnchantedBookItem.createForEnchantment(new EnchantmentInstance(
+                getEnchantment(enchantmentId),
+                level
+            ))),
             Ingredient.of(new ItemStack(Items.INSTANCE.getQuartzEnrichedIron())),
             Ingredient.of(new ItemStack(Blocks.BOOKSHELF)),
             Ingredient.of(new ItemStack(Items.INSTANCE.getUpgrade())),
@@ -60,26 +62,25 @@ public class UpgradeWithEnchantedBookRecipe extends ShapedRecipe {
             Ingredient.of(new ItemStack(Items.INSTANCE.getQuartzEnrichedIron())),
             Ingredient.of(new ItemStack(Items.INSTANCE.getQuartzEnrichedIron())),
             Ingredient.of(new ItemStack(Items.INSTANCE.getQuartzEnrichedIron()))
-        ), Optional.empty()), new ItemStack(resultItem.value()));
-        this.enchantment = enchantment;
+        ), Optional.empty()), new ItemStack(BuiltInRegistries.ITEM.get(resultItemId)));
+        this.enchantmentId = enchantmentId;
         this.level = level;
-        this.resultItem = resultItem;
+        this.resultItemId = resultItemId;
     }
 
-    public Holder<Item> getResultItem() {
-        return resultItem;
+    private static Enchantment getEnchantment(final ResourceLocation enchantmentId) {
+        return requireNonNull(BuiltInRegistries.ENCHANTMENT.get(enchantmentId));
     }
 
-    public Holder<Enchantment> getEnchantment() {
-        return enchantment;
+    ResourceLocation getResultItemId() {
+        return resultItemId;
     }
 
-    @Nullable
-    public ResourceLocation getEnchantmentId() {
-        return enchantment.unwrapKey().map(ResourceKey::location).orElse(null);
+    ResourceLocation getEnchantmentId() {
+        return enchantmentId;
     }
 
-    public int getEnchantmentLevel() {
+    int getEnchantmentLevel() {
         return level;
     }
 

--- a/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/upgrade/UpgradeWithEnchantedBookRecipeSerializer.java
+++ b/refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/upgrade/UpgradeWithEnchantedBookRecipeSerializer.java
@@ -1,16 +1,9 @@
 package com.refinedmods.refinedstorage2.platform.common.upgrade;
 
-import java.util.Objects;
-
 import com.mojang.serialization.Codec;
-import net.minecraft.core.Holder;
-import net.minecraft.core.registries.BuiltInRegistries;
-import net.minecraft.core.registries.Registries;
 import net.minecraft.network.FriendlyByteBuf;
-import net.minecraft.resources.ResourceKey;
-import net.minecraft.world.item.Item;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.crafting.RecipeSerializer;
-import net.minecraft.world.item.enchantment.Enchantment;
 
 public class UpgradeWithEnchantedBookRecipeSerializer implements RecipeSerializer<UpgradeWithEnchantedBookRecipe> {
     @Override
@@ -20,22 +13,16 @@ public class UpgradeWithEnchantedBookRecipeSerializer implements RecipeSerialize
 
     @Override
     public UpgradeWithEnchantedBookRecipe fromNetwork(final FriendlyByteBuf buf) {
-        final Holder<Item> result = BuiltInRegistries.ITEM.getHolder(ResourceKey.create(
-            Registries.ITEM,
-            buf.readResourceLocation()
-        )).orElseThrow();
-        final Holder<Enchantment> enchantment = BuiltInRegistries.ENCHANTMENT.getHolder(ResourceKey.create(
-            Registries.ENCHANTMENT,
-            buf.readResourceLocation()
-        )).orElseThrow();
+        final ResourceLocation resultItemId = buf.readResourceLocation();
+        final ResourceLocation enchantmentId = buf.readResourceLocation();
         final int level = buf.readInt();
-        return new UpgradeWithEnchantedBookRecipe(enchantment, level, result);
+        return new UpgradeWithEnchantedBookRecipe(enchantmentId, level, resultItemId);
     }
 
     @Override
     public void toNetwork(final FriendlyByteBuf buf, final UpgradeWithEnchantedBookRecipe recipe) {
-        buf.writeResourceLocation(recipe.getResultItem().unwrapKey().orElseThrow().location());
-        buf.writeResourceLocation(Objects.requireNonNull(recipe.getEnchantmentId()));
+        buf.writeResourceLocation(recipe.getResultItemId());
+        buf.writeResourceLocation(recipe.getEnchantmentId());
         buf.writeInt(recipe.getEnchantmentLevel());
     }
 }

